packages/typescript-eslint/src/index.ts

```diff
@@ -120,7 +122,35 @@ export const configs = {
    */
   stylisticTypeCheckedOnly:
     rawPlugin.flatConfigs['flat/stylistic-type-checked-only'],
-};
+});
+
+function createConfigsGetters<T extends object>(values: T): T {
+  const configs = {};
+
+  Object.defineProperties(
+    configs,
+    Object.fromEntries(
+      Object.entries(values).map(([key, value]: [string, unknown]) => [
+        key,
+        {
+          get: () => {
+            const candidateRootDir = getRootDirFromStack(
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              new Error().stack!,
+            );
+            if (candidateRootDir) {
+              addCandidateTSConfigRootDir(candidateRootDir);
```

Comment:
[Testing] TODO: Add unit tests for this. `expect(...).toHaveBeenCalledWith()` and such.